codespell_lib/tests/test_dictionary.py

```diff
@@ -169,21 +171,78 @@ def test_error_checking_in_aspell(err, rep, err_aspell, rep_aspell, match):
         _check_err_rep(err, rep, (err_aspell, rep_aspell), 'dummy')
 
 
+# allow some duplicates, like "m-i-n-i-m-i-s-e", or "c-a-l-c-u-l-a-t-a-b-l-e"
+allowed_dups = {
+    ('dictionary.txt', 'dictionary_en-GB_to_en-US.txt'),
+    ('dictionary.txt', 'dictionary_rare.txt'),
+}
+
+
 @fname_params
+@pytest.mark.dependency(name='dictionary loop')
 def test_dictionary_looping(fname, in_aspell):
     """Test that all dictionary entries are valid."""
-    err_dict = dict()
+    this_err_dict = dict()
+    short_fname = op.basename(fname)
     with open(fname, 'rb') as fid:
         for line in fid:
             err, rep = line.decode('utf-8').split('->')
             err = err.lower()
-            assert err not in err_dict, 'error %r already exists' % err
+            assert err not in this_err_dict, \
+                'error %r already exists in %s' % (err, short_fname)
             rep = rep.rstrip('\n')
             reps = [r.strip() for r in rep.lower().split(',')]
             reps = [r for r in reps if len(r)]
-            err_dict[err] = reps
-    # check for corrections that are errors (but not self replacements)
-    for err in err_dict:
-        for r in err_dict[err]:
-            assert (r not in err_dict) or (r in err_dict[r]), \
-                ('error %s: correction %s is an error itself' % (err, r))
+            this_err_dict[err] = reps
+    # 1. check the dict against itself (diagonal)
+    for err in this_err_dict:
+        for r in this_err_dict[err]:
+            assert r not in this_err_dict, \
+                ('error %s: correction %s is an error itself in the same '
+                 'dictionary file %s' % (short_fname, err, r))
+    pair = (short_fname, short_fname)
+    assert pair not in global_pairs
+    global_pairs.add(pair)
+    for other_fname, other_err_dict in global_err_dicts.items():
+        # error duplication (eventually maybe we should just merge?)
+        for err in this_err_dict:
+            assert err not in other_err_dict, \
```

Comment:
This isn't doing your clever assertion error gathering either, so it currently fails on the first one, I assume, (and also says it hasn't run all the tests):
https://travis-ci.org/github/codespell-project/codespell/jobs/687269384

Reply:
Indeed it's not, feel free to add it if you want (it can always be added later)

Reply:
I'll leave it for later for now.